lib/usb/usb_fx07_common.c

```diff
@@ -342,3 +336,12 @@ void stm32fx07_disconnect(usbd_device *usbd_dev, bool disconnected)
 		REBASE(OTG_DCTL) &= ~OTG_FS_DCTL_SDIS;
 	}
 }
+
+void stm32fx07_register_sof_callback(usbd_device *usbd_dev)
```

Comment:
I like what you're doing with this PR, but I don't think the naming is right.  The outer layer registers a callback, the innerlayer is enabling sof interrupts.